nautobot/extras/models/relationships.py

```diff
@@ -32,6 +32,19 @@ class RelationshipModel(models.Model):
     class Meta:
         abstract = True
 
+    # Define GenericRelations so that deleting a RelationshipModel instance
+    # cascades to deleting any RelationshipAssociations that were using this instance
+    source_for_associations = GenericRelation(
+        "extras.RelationshipAssociation",
+        content_type_field="source_type",
+        object_id_field="source_id",
+    )
+    destination_for_associations = GenericRelation(
+        "extras.RelationshipAssociation",
+        content_type_field="destination_type",
+        object_id_field="destination_id",
+    )
```

Comment:
And by "a lot of new related names", I meant:

```
(Pdb) RelationshipAssociation.objects.filter(source_potato=None)
*** django.core.exceptions.FieldError: Cannot resolve keyword 'source_potato' into field. Choices are: destination, destination_aggregate, destination_cable, destination_circuit, destination_circuittermination, destination_circuittype, destination_cluster, destination_clustergroup, destination_clustertype, destination_consoleport, destination_consoleporttemplate, destination_consoleserverport, destination_consoleserverporttemplate, destination_device, destination_devicebay, destination_devicebaytemplate, destination_devicerole, destination_devicetype, destination_exporttemplate, destination_frontport, destination_frontporttemplate, destination_gitrepository, destination_id, destination_interface, destination_interfacetemplate, destination_inventoryitem, destination_ipaddress, destination_manufacturer, destination_platform, destination_powerfeed, destination_poweroutlet, destination_poweroutlettemplate, destination_powerpanel, destination_powerport, destination_powerporttemplate, destination_prefix, destination_provider, destination_rack, destination_rackgroup, destination_rackreservation, destination_rackrole, destination_rearport, destination_rearporttemplate, destination_region, destination_rir, destination_role, destination_routetarget, destination_service, destination_site, destination_status, destination_tag, destination_tenant, destination_tenantgroup, destination_type, destination_type_id, destination_virtualchassis, destination_virtualmachine, destination_vlan, destination_vlangroup, destination_vminterface, destination_vrf, id, relationship, relationship_id, source, source_aggregate, source_cable, source_circuit, source_circuittermination, source_circuittype, source_cluster, source_clustergroup, source_clustertype, source_consoleport, source_consoleporttemplate, source_consoleserverport, source_consoleserverporttemplate, source_device, source_devicebay, source_devicebaytemplate, source_devicerole, source_devicetype, source_exporttemplate, source_frontport, source_frontporttemplate, source_gitrepository, source_id, source_interface, source_interfacetemplate, source_inventoryitem, source_ipaddress, source_manufacturer, source_platform, source_powerfeed, source_poweroutlet, source_poweroutlettemplate, source_powerpanel, source_powerport, source_powerporttemplate, source_prefix, source_provider, source_rack, source_rackgroup, source_rackreservation, source_rackrole, source_rearport, source_rearporttemplate, source_region, source_rir, source_role, source_routetarget, source_service, source_site, source_status, source_tag, source_tenant, source_tenantgroup, source_type, source_type_id, source_virtualchassis, source_virtualmachine, source_vlan, source_vlangroup, source_vminterface, source_vrf
```

Reply:
Yeah, I'd initially done this and then got cold feet when I realized how many additional filters it would add. Given that only two of them will even apply to any given RelationshipAssociation instance, does that add too much noise in general? Or do you think the benefit outweighs that?

Reply:
IMO this "noise" is a problem since it's only there if you need to use it and otherwise completely invisible. I am of the mind that the benefit in having a reliable forward/reverse relationship like regular FK/M2M is worth it.

Reply:
Ok, stand by for updates.